sdk/python/lib/pulumi/dynamic/__main__.py

```diff
@@ -34,10 +34,37 @@
 _GRPC_CHANNEL_OPTIONS = [("grpc.max_receive_message_length", _MAX_RPC_MESSAGE_SIZE)]
 
 
-def get_provider(props) -> ResourceProvider:
-    byts = base64.b64decode(props[PROVIDER_KEY])
-    return dill.loads(byts)
+#
+#const providerCache: { [key: string]: dynamic.ResourceProvider } = {};
+#
+#function getProvider(props: any): dynamic.ResourceProvider {
+#    const providerString = props[providerKey];
+#    let provider: any = providerCache[providerString];
+#    if (!provider) {
+#        provider = requireFromString(providerString).handler();
+#        providerCache[providerString] = provider;
+#    }
+#
+#    // TODO[pulumi/pulumi#414]: investigate replacing requireFromString with eval
+#    return provider;
+#}
```

Comment:
Is there a particular reason to keep this comment here?  It feels like this is likely to get out of sync over time anyway.

Reply:
herp no, I'll clean this up